.claude/session_context/docs/comprehensive_code_review.md

```
@@ -0,0 +1,255 @@
+   - **Action**: Keep as-is, provides good value
+
+## Major Areas of Concern
+
+### Critical: Resource Management Edge Cases
+
+**Location**: `internal/parser/parser.go:131-143`
+**Issue**: HTTP client creation logic creates different client types in different scenarios
+```go
+// This logic is complex and error-prone:
+if opts.HTTPClient != nil {
+    httpClient = &resource.HTTPClient{
```

markdownlint-cli2 (0.17.2)

Lists should be surrounded by blank lines

(MD032, blanks-around-lists)